test/Tests/src/Data/XML/XML_Spec.enso

```diff
@@ -0,0 +1,175 @@
+
+            root.at 0 . attribute "id" . should_equal "100"
+            root.at 1 . attribute "id" . should_equal "101"
+            root.at 2 . attribute "studentId" . should_equal "1000"
+            root.at 3 . attribute "studentId" . should_equal "1001"
+
+            root.at 3 . attribute "does_not_exist" if_missing="if_missing" . should_equal "if_missing"
+
+        Test.specify "Can get element an attribute map" <|
+            root.at 2 . attributes . should_equal (Map.from_vector [["studentId", "1000"], ["year", "2"]])
+            root.at 3 . attributes . should_equal (Map.from_vector [["studentId", "1001"], ["year", "3"]])
+
+        Test.specify "Can get nodes via xpath" <|
+            classes = root.at "/class"
+            classes.length . should_equal 1
+            classes.at 0 . name . should_equal "class"
+
+            teachers = root.at "/class/teacher"
+            teachers.length . should_equal 2
+            teachers.at 0 . at "@id" . should_equal "100"
+            teachers.at 1 . at "@id" . should_equal "101"
+
+            students = root.at "/class/student"
+            students.length . should_equal 3
+            students.at 0 . at "@studentId" . should_equal "1000"
+            students.at 1 . at "@studentId" . should_equal "1001"
+
+            root.at "/class/teacher[1]/firstname" . at 0 . text . should_equal "Mary"
+            root.at "/class/teacher[2]/firstname" . at 0 . text . should_equal "Bob"
+            root.at "/class/teacher[1]/firstname/text()" . should_equal ["Mary"]
+            root.at "/class/teacher[2]/firstname/text()" . should_equal ["Bob"]
+            root.at "/class/teacher/firstname/text()" . should_equal ["Mary", "Bob"]
+            root.at "/class/teacher[1]/bio" . at 0 . text . should_equal '\n            Blah blah\n        '
+            root.at "/class/teacher[2]/bio" . at 0 . text . should_equal '\n            This that\n        '
+            root.get "/class/teacher[23]" . should_equal []
+
+            root.at "teacher[1]/firstname" . at 0 . text . should_equal "Mary"
+            root.at "teacher[2]/firstname" . at 0 . text . should_equal "Bob"
+            root.at "teacher[1]/bio" . at 0 . text . should_equal '\n            Blah blah\n        '
+            root.at "teacher[2]/bio" . at 0 . text . should_equal '\n            This that\n        '
+
+        Test.specify "get" <|
+            root.get 0 . get 0 . name . should_equal "firstname"
+            root.get 0 . get "@id" . should_equal "100"
+            root.get "/class/teacher[1]/firstname" . get 0 . text . should_equal "Mary"
+
+            root.get 0 . get 32 "if_missing" . should_equal "if_missing"
+            root.get 0 . get "@not_there" "if_missing" . should_equal "if_missing"
+
+        Test.specify "Can handle a bad xpath" <|
+            root.at "/qqq[[[[1" . at 0 . text . should_fail_with XML_Error
+
+    Test.group "children" <|
+        Test.specify "children" <|
```

Comment:
The group and test case names are very vague and do not say anything about what is being tested really. Please make them clearer.

Reply:
Done, here and elsewhere.